### PR TITLE
[FileSystem Dock] Add symlink indicator and tooltip.

### PR DESCRIPTION
**Pull request by bruvzg**
Adds symlink arrow indicator and tooltip with link source.

Screenshot 2024-09-06 at 11 00 01

See https://github.com/godotengine/godot/pull/96603#pullrequestreview-2284078638

### CODE REVIEW

File: doc/classes/TreeItem.xml [hunk 1]
```diff
@@ -662,6 +669,14 @@
 				Modulates the given column's icon with [param modulate].
 			</description>
 		</method>
+		<method name="set_icon_overlay">
```
Comment (Maran23) [inline doc/classes/TreeItem.xml:672]: I wonder if we should just change the icon instead of exposing another new method in `Tree`.

Comment (bruvzg) [reply doc/classes/TreeItem.xml:672]: Symlink can be any file or folder with different icons, changing icon would require generating a new icon for each.

**Comment by passivestar** [comment]
Idea for another PR: Create symlinks instead of copying if you hold <kbd>Alt+Ctrl/Cmd</kbd> when dropping files onto the dock

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Hilderin** [comment]
I have some concern about this PR because it adds a check directly on disk for each directory and file everytime the FileSystemDock tree is recreated which is quite often. That will slow down considerably the tree update with large projects. It's already very slow for project with couple of thousand file (like 2-5sec). With a roundtrip to the disk, that will probably be catastrophic. In my opinion, the link information should have been added in the EditorFileSystem file tree and read while scanning and updated based on the last modified time like the other path and file properties. I did not do testing to prove my concern. I'll be glad to test it further and implement it if you wish.

**Comment by akien-mga** [comment]
Yeah that's a very good point, would be worth testing further to confirm the performance impact and how to improve it.

**Pull request merged**